lib/spack/spack/build_environment.py

```diff
@@ -993,8 +994,24 @@ def add_modifications_for_dep(dep):
             dpkg = dep.package
             if set_package_py_globals:
                 set_module_variables_for_package(dpkg)
+
             # Allow dependencies to modify the module
-            dpkg.setup_dependent_package(spec.package.module, spec)
+            # Get list of modules that may need updating
+            modules = []
+            for cls in inspect.getmro(type(spec.package)):
+                module = cls.module
+                if module == spack.package_base:
+                    break
+                modules.append(module)
+
+            # Execute changes as if on a single module
+            # copy dict to ensure prior changes are available
+            changes = spack.util.pattern.Bunch()
+            dpkg.setup_dependent_package(changes, spec)
```

Comment:
I don't think many packages would be affected, but this potentially breaks previous semantics since we are not passing something that resembles the module. So if a package was accessing module properties, it won't find them in the `Bunch()`. Also, the `PackageBase` docstring is now misleading, since it says we are passing `package.module` as an argument.